lib/Dialect/Torch/Transforms/RefineTypes.cpp

```diff
@@ -1314,21 +1320,37 @@ ChangeResult TypeAnalyzer::visitAtenEmbeddingOp(
   return getLatticeElement(op.getResult()).join(knowledge);
 }
 
+template <typename OpTy>
+static ValueKnowledge
+getSameSizeAsInput(OpTy op,
```

Comment:
nit: You don't need a template here. Since the body of the function does not use any functions of the form `op.function(...)`, then the code also works with `Operation *op` as an input.